core/src/com/mygdx/game/models/HexMap.java

```diff
@@ -48,31 +50,35 @@ public static HexMap getHexMapFromConfig(String content) {
         // Set up units
         int row = 0;
         int column = 0;
-        for (JsonElement j : jsonArrayUnits1) { // loop through outer Array
-            List<Unit> unitsList = new ArrayList<>();
-            JsonArray jsonArrayUnits2 = j.getAsJsonArray();
-            for (JsonElement k : jsonArrayUnits2) { // loop through inner Array
-                String text = k.getAsString(); // "archer.json", "soldier.json", "null", etc.
-                Unit unit0 = text.equals("null") ? null : cf.makeUnitFromConfig(text);
-                unitsList.add(unit0);
-                units.setHex(new Position(row, column), unitsList); // set up the HexBoard
-                column++;
+        for (JsonElement j : jsonArrayUnits1) {
+            String config = j.getAsJsonObject().get("config").getAsString(); // "archer.json", "null", etc.
+            int id = j.getAsJsonObject().get("id").getAsInt();
+            int pid = j.getAsJsonObject().get("pid").getAsInt();
+            BasicUnit unit0 = config.equals("null") ? null : cf.makeUnitFromConfig(config, id, pid);
+            List<BasicUnit> unitsList = new ArrayList<>();
+            unitsList.add(unit0);
+            units.setHex(new Position(row, column), unitsList); // set up the HexBoard
+            column++;
+            if (column == columns) {
+                column = 0;
+                row++;
             }
-            row++;
         }
 
         // Set up terrain
         row = 0;
         column = 0;
         for (JsonElement j : jsonArrayTerrain1) {
-            JsonArray jsonArrayTerrain2 = j.getAsJsonArray();
-            for (JsonElement k : jsonArrayTerrain2) {
-                String text = k.getAsString(); // "swamp.json", "desert.json", "ocean.json", "snow.json", etc.
-                Terrain terrain0 = text.equals("null") ? null : cf.makeTerrainFromConfig(text);
-                terrain.setHex(new Position(row, column), terrain0);
-                column++;
+            String config = j.getAsJsonObject().get("config").getAsString(); // "swamp.json", "desert.json", etc.
+            String texture = j.getAsJsonObject().get("texture").getAsString();
+            int id = j.getAsJsonObject().get("id").getAsInt();
+            Terrain terrain0 = config.equals("null") ? null : cf.makeTerrainFromConfig(config, texture, id);
+            terrain.setHex(new Position(row, column), terrain0);
+            column++;
+            if (column == columns) {
```

Comment:
If we need the index in the arraay we should just change the for loops to increment an integer.